### PR TITLE
Gradle6 

### PR DESCRIPTION
**Pull request by rooeque**
Update to Gradle 6. 

Deprecations on our side have been resolved. However, there are new deprecations against Gradle 7.x in the `scalastyle` and `scala-multiversion` plugin. See here: https://scans.gradle.com/s/hpux57wxwe2yg/deprecations

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -56,34 +56,31 @@ targetCompatibility = 1.8
...
 // keep in alphabetical order
 dependencies {
-    compile "com.google.code.gson:gson:${versions.gson}"
-    compile "com.google.guava:guava:${versions.guava}"
-    compile "com.google.inject:guice:${versions.guice}"
-    compile "com.typesafe:config:${versions.typesafeConfig}"
-    compile "com.typesafe.slick:slick_%%:${versions.slick}"
-    compile "com.typesafe.slick:slick-hikaricp_%%:${versions.slick}"
-    compile "commons-io:commons-io:${versions.commonsIo}"
+    implementation "com.google.code.gson:gson:${versions.gson}"
```
Comment (rooeque) [inline build.gradle]: `compile` and `runtime` have been deprecated in favor of `implementation`: 

https://docs.gradle.org/6.0/userguide/upgrading_version_5.html#changes_6.0

File: build.gradle [hunk 2]
```diff
@@ -56,34 +56,31 @@ targetCompatibility = 1.8
...
-    compile "org.scala-lang:scala-reflect:%scala-version%"
-    compile "org.scalactic:scalactic_%%:${versions.scalactic}"
-    compile "org.scalatest:scalatest_%%:${versions.scalatest}"
-    compile "org.tinylog:tinylog:${versions.tinylog}"
-
-    compile files(org.gradle.internal.jvm.Jvm.current().toolsJar)
-
-    testCompile "io.gatling:gatling-commons:${versions.gatling}"
-
-    scoverage "org.scoverage:scalac-scoverage-plugin_%%:${versions.scalacScoverage}"
```
Comment (rooeque) [inline build.gradle]: no longer need to declare this in newer scoverage version: https://github.com/scoverage/gradle-scoverage#migration-to-3x

File: build.gradle [hunk 3]
```diff
@@ -125,19 +122,19 @@ processResources {
 }
 
 task sourcesJar(type: Jar) {
-    classifier = 'sources'
+    archiveClassifier = 'sources'
```
Comment (rooeque) [inline build.gradle]: `classifier` has been deprecated in favor of `archiveClassifier`

File: build.gradle [hunk 4]
```diff
@@ -176,16 +173,6 @@ test {
     }
 }
 
-testScoverage {
```
Comment (rooeque) [inline build.gradle]: `test` configuration will now run scoverage whenever task is invoked: https://github.com/scoverage/gradle-scoverage#migration-to-3x

File: versionInfo.gradle [hunk 5]
```diff
@@ -38,10 +38,10 @@ project.ext.versions = [
     , junit: '4.12'
     , math3: '3.6.1'
     , mysqlConnector: '8.0.16'
-    , nebulaDependencyLock: '5.0.0'
-    , nebulaGit: '3.0.1'
-    , nebulaRelease: '6.3.5'
-    , nebulaPublish: '8.2.0'
+    , nebulaDependencyLock: '8.3.0'
+    , nebulaGit: '4.2.0'
```
Comment (rooeque) [inline versionInfo.gradle]: NICE

File: build.gradle (hunk 1 above)
Comment (tomnis) [reply build.gradle]: 👍

File: build.gradle (hunk 2 above)
Comment (tomnis) [reply build.gradle]: 👍

File: versionInfo.gradle (hunk 5 above)
Comment (tomnis) [reply versionInfo.gradle]: :P

**Comment by tomnis** [comment]
cool. why was the lockfile removed? are we using gradle native locking mechanism now?

**Comment by rooeque** [comment]
the lock file is still there, I think the structure changed a little bit with the gradle upgrade.

**Pull request merged**